src/clzones.cpp

```diff
@@ -582,16 +647,47 @@ bool zone_manager::has_loot_dest_near( const tripoint &where ) const
     return false;
 }
 
+const zone_data *zone_manager::get_zone_at( const tripoint &where, const zone_type_id &type ) const
+{
+    for( auto it = zones.rbegin(); it != zones.rend(); ++it ) {
+        const auto &zone = *it;
+
+        if( zone.has_inside( where ) && zone.get_type() == type ) {
+            return &zone;
+        }
+    }
+    return nullptr;
+}
+
+bool zone_manager::custom_loot_has( const tripoint &where, const item *item_type ) const
+{
+    auto zone = get_zone_at( where, zone_type_id( "LOOT_CUSTOM" ) );
+    if( !zone || !item_type ) {
+        return false;
+    }
+    const loot_options options = dynamic_cast<const loot_options &>( zone->get_options() );
+    std::string filter_string = options.get_mark();
+    auto z = item_filter_from_string( filter_string );
+
+    return z( *item_type );
+}
+
 std::unordered_set<tripoint> zone_manager::get_near( const zone_type_id &type,
-        const tripoint &where, int range, const faction_id &fac ) const
+        const tripoint &where, int range, const item *item_type, const faction_id &fac ) const
```

Comment:
Isn't that parameter more like an actual item and not an item type? The name should match that.